src/settings.ml

```diff
@@ -17,9 +17,7 @@ let get ?section t =
     match t.ofJson v with
     | s -> Some s
     | exception Json.Decode.DecodeError msg ->
-      let (_ : unit Promise.t) =
-        Window.showErrorMessage (sprintf "Setting %s is invalid: %s" t.key msg)
-      in
+      message `Error "Setting %s is invalid: %s" t.key msg;
```

Comment:
Actually, I think this change is too drastic. How would the user know if he accidentally typod the setting for example?

How about we just ignore nulls and show the error when a legit parsing error occurs.

Reply:
It does ignore nulls. `None` (`null`) is returned from the function without showing an error message here:
https://github.com/ocamllabs/vscode-ocaml-platform/blob/17100c6531beb97a117c45654962e09d6c596f84/src/settings.ml#L15

If the user typod the setting and it does not exist, `Settings.get` returns `None` without showing an error message, as it was originally intended. This PR is just an implementation fix because the previous version expected `null` to turn into ` None` `option` automatically, but it didn't.

All of the other functions that receive a value from `Settings.get` are already set up to handle the situation if it returns `None`, which happens when the setting doesn't exist.

Reply:
Oh ok. For some reason I thought `message` just displays it in the log.